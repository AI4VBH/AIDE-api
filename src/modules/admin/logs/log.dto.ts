export class LogsDTO {
  json: LogDTO;

  public static from(dto: Partial<LogsDTO>) {
    const it = new LogsDTO();
    it.json = dto.json;
    return it;
  }

  // TODO: Implement fromEntity method
  // public static fromEntity(entity: Logs) {
  //   return this.from({
  //     task_id: entity.id,
  //   });
  // }

  // TODO: Implement toEntity method
  // public toEntity(logs: Logs = null) {
  //   const it = new Logs();
  //   it.task_id = this.task_id;
  //   return it;
  // }
}

export class LogDTO {
  execution_id: string;
  level: string;
  line_no: number;
  logger: string;
  model_name: string;
  model_version: string;
  module: string;
  msg: string;
  thread: string;
  type: string;
  written_at: string;
  written_ts: number;

  public static from(dto: Partial<LogDTO>) {
    const it = new LogDTO();
    it.execution_id = dto.execution_id;
    it.level = dto.level;
    it.line_no = dto.line_no;
    it.logger = dto.logger;
    it.model_name = dto.model_name;
    it.model_version = dto.model_version;
    it.module = dto.module;
    it.msg = dto.msg;
    it.thread = dto.thread;
    it.type = dto.type;
    it.written_at = dto.written_at;
    it.written_ts = dto.written_ts;
    return it;
  }

  // TODO: Implement fromEntity method
  // public static fromEntity(entity: Log) {
  //   return this.from({
  //     task_id: entity.id,
  //   });
  // }

  // TODO: Implement toEntity method
  // public toEntity(log: Log = null) {
  //   const it = new Log();
  //   it.task_id = this.task_id;
  //   return it;
  // }
}
