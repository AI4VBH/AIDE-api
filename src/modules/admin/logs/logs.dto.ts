export class LogsDto {
  public renderedMessage: string;
  public timestamp: Date;
  public level: string;
}
