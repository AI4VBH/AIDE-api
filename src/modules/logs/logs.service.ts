import { Injectable } from '@nestjs/common';
import { Log } from './log.interface';

@Injectable()
export class LogsService {
  getLog(): Log {
    return {} as Log;
  }
}
