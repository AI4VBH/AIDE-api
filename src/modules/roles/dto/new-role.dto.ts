import { IsNotEmpty } from 'class-validator';

export class NewRoleDto {
  @IsNotEmpty()
  name: string;

  description?: string;
}
