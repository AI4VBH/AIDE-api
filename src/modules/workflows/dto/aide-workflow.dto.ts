/*
 * Copyright 2022 Crown Copyright
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */

export interface PagedWorkflowsDto {
  totalPages: number;
  totalRecords: number;
  data: PagedWorkflowsItemDto[];
}

export interface PagedWorkflowsItemDto {
  id: string;
  workflow_id: string;
  revision: number;
  name: string;
  version: string;
  description: string;
  ae_title: string;
  data_origins: string[];
}

interface InformaticsGateway {
  ae_title: string;
  export_destinations: string[];
}

export interface WorkflowDto extends Object {
  [key: string]: any;
  informatics_gateway: Partial<InformaticsGateway>;
}

export interface CreateEditWorkflowDto {
  workflow: Partial<WorkflowDto>;
}
