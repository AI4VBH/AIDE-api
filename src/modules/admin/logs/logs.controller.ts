/*
 * Copyright 2022 Guy’s and St Thomas’ NHS Foundation Trust
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */

import {
  BadRequestException,
  Controller,
  Get,
  Param,
  UseFilters,
} from '@nestjs/common';
import ExternalServerExceptionFilter from 'shared/http/external-server-exception.filter';
import { LogsService } from './logs.service';

@Controller('logs')
@UseFilters(ExternalServerExceptionFilter)
export class LogsController {
  constructor(private readonly appService: LogsService) {}

  @Get(':id')
  getLogs(@Param('id') id: string) {
    if (!id) {
      throw new BadRequestException('id param cannot be empty');
    }
    return this.appService.getLogByTask(id);
  }
}
