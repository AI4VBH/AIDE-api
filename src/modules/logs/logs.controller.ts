import { Controller, Get } from '@nestjs/common';
import { Log } from './log.interface';
import { LogsService } from './logs.service';

@Controller('logs')
export class LogsController {
  constructor(private readonly appService: LogsService) {}

  @Get()
  getHello(): Log {
    return this.appService.getLog();
  }
}
