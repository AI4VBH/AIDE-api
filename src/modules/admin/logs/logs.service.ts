import { Inject, Injectable } from '@nestjs/common';
import { ElasticClient } from 'shared/elastic/elastic-client';
import { LogsDto } from './logs.dto';
import { IElasticLogObject } from './models/logs.interfaces';

@Injectable()
export class LogsService {
  @Inject(ElasticClient)
  private readonly elasticClient: ElasticClient;

  async getLogByTask(id: string): Promise<LogsDto[]> {
    const response = await this.elasticClient.getLogs(id);

    if (response.statusCode != 200 || response.body.hits.total.value == 0) {
      return [];
    }

    const body = response.body as IElasticLogObject;

    const dtoArr: LogsDto[] = [];
    for (const hit of body.hits.hits) {
      dtoArr.push({
        level: hit._source.Level,
        renderedMessage: hit._source.RenderedMessage,
        timestamp: hit._source.Timestamp,
      });
    }

    return dtoArr;
  }
}
