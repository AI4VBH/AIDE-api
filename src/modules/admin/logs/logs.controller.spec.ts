import { createMock, DeepMocked } from '@golevelup/ts-jest';
import { BadRequestException } from '@nestjs/common';
import { Test, TestingModule } from '@nestjs/testing';
import { LogsController } from './logs.controller';
import { LogsDto } from './logs.dto';
import { LogsService } from './logs.service';

describe('LogsController', () => {
  let controller: LogsController;
  let logsService: DeepMocked<LogsService>;

  beforeEach(async () => {
    logsService = createMock<LogsService>();

    const module: TestingModule = await Test.createTestingModule({
      controllers: [LogsController],
      providers: [
        {
          provide: LogsService,
          useValue: logsService,
        },
      ],
    }).compile();

    controller = module.get<LogsController>(LogsController);
  });

  it('should be defined', () => {
    expect(controller).toBeDefined();
  });

  describe('get logs', () => {
    it('calls service', async () => {
      const task = 'lillie-task-1';
      const expectedDto: LogsDto[] = [
        {
          level: 'Information',
          renderedMessage: 'this is a message',
          timestamp: new Date('2022-09-26T15:19:22.9668840Z'),
        },
        {
          level: 'Information',
          renderedMessage: 'this is a message',
          timestamp: new Date('2022-09-26T15:19:22.9668840Z'),
        },
        {
          level: 'Debug',
          renderedMessage:
            'Publishing message to "rabbitmq.monai"/"monaideploy". Exchange="monaideploy", Routing Key="md.tasks.dispatch".',
          timestamp: new Date('2022-09-26T15:18:22.9668840Z'),
        },
      ];

      logsService.getLogByTask.mockResolvedValue(expectedDto);

      const response = await controller.getLogs(task);

      expect(response).toMatchSnapshot();
      expect(logsService.getLogByTask).toHaveBeenCalledWith(task);
    });

    it.each([null, undefined])('id param cannot be empty', async (id) => {
      const action = async () => await controller.getLogs(id);

      await expect(action).rejects.toThrow(BadRequestException);
    });
  });
});
