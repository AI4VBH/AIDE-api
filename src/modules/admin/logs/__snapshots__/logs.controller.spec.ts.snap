// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogsController get logs calls service 1`] = `
[
  {
    "level": "Information",
    "renderedMessage": "this is a message",
    "timestamp": 2022-09-26T15:19:22.966Z,
  },
  {
    "level": "Information",
    "renderedMessage": "this is a message",
    "timestamp": 2022-09-26T15:19:22.966Z,
  },
  {
    "level": "Debug",
    "renderedMessage": "Publishing message to "rabbitmq.monai"/"monaideploy". Exchange="monaideploy", Routing Key="md.tasks.dispatch".",
    "timestamp": 2022-09-26T15:18:22.966Z,
  },
]
`;
